aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1719 @@
+                amount_added: amount,
+                add_stake_fee,
+            },
+        );
+    }
+
+    /// Unlock `amount` from the active + pending_active stake of `delegator` or
+    /// at most how much active stake there is on the stake pool.
+    public entry fun unlock(delegator: &signer, pool_address: address, amount: u64) acquires DelegationPool {
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+        // short-circuit if amount to unlock is 0 so no event is emitted
+        if (amount == 0) { return };
+
+        let pool = borrow_global_mut<DelegationPool>(pool_address);
+        let delegator_address = signer::address_of(delegator);
+
+        // capture how much stake would be unlocked on the stake pool
+        let (active, _, _, _) = stake::get_stake(pool_address);
+        amount = min(amount, active);
+        amount = redeem_active_shares(pool, delegator_address, amount);
+
+        stake::unlock(&retrieve_stake_pool_signer(pool), amount);
+        buy_in_inactive_shares(pool, delegator_address, amount);
+
+        event::emit_event(
+            &mut pool.unlock_stake_events,
+            UnlockStakeEvent {
+                pool_address,
+                delegator_address,
+                amount_unlocked: amount,
+            },
+        );
+    }
+
+    /// Move `amount` of coins from pending_inactive to active.
+    public entry fun reactivate_stake(delegator: &signer, pool_address: address, amount: u64) acquires DelegationPool {
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+        // short-circuit if amount to reactivate is 0 so no event is emitted
+        if (amount == 0) { return };
+
+        let pool = borrow_global_mut<DelegationPool>(pool_address);
+        let delegator_address = signer::address_of(delegator);
+
+        let observed_lockup_cycle = pool.observed_lockup_cycle;
+        amount = redeem_inactive_shares(pool, delegator_address, amount, observed_lockup_cycle);
+
+        stake::reactivate_stake(&retrieve_stake_pool_signer(pool), amount);
+        buy_in_active_shares(pool, delegator_address, amount);
+
+        event::emit_event(
+            &mut pool.reactivate_stake_events,
+            ReactivateStakeEvent {
+                pool_address,
+                delegator_address,
+                amount,
+            },
+        );
+    }
+
+    /// Withdraw `amount` of owned inactive stake from the delegation pool at `pool_address`.
+    public entry fun withdraw(delegator: &signer, pool_address: address, amount: u64) acquires DelegationPool {
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+        withdraw_internal(borrow_global_mut<DelegationPool>(pool_address), signer::address_of(delegator), amount);
+    }
+
+    #[view]
```

Comment:
Can we move all the view functions to the top, after the struct definitions but before any non-view functions? It's the place we're been trying to keep view functions in other modules for consistency.